File: elixir/lib/formhandle/prompt.ex
defmodule FormHandle.Prompt do
  @moduledoc "Interactive prompt helpers."

  def ask(question) do
    case IO.gets(question) do
      :eof ->
        IO.puts("")
        System.halt(1)

      {:error, _} ->
        IO.puts("")
        System.halt(1)

      answer ->
        String.trim(answer)
    end
  end

  def confirm(question) do
    answer = ask("#{question} (y/N) ")
    String.downcase(answer) in ["y", "yes"]
  end
end

File: ruby/lib/formhandle/prompt.rb
# frozen_string_literal: true

module FormHandle
  module Prompt
    def self.ask(question)
      print question
      $stdout.flush
      answer = $stdin.gets
      exit 1 if answer.nil?
      answer.strip
    end

    def self.confirm(question)
      answer = ask("#{question} (y/N) ")
      %w[y yes].include?(answer.downcase)
    end
  end
end
